### PR TITLE
[SYCL] Remove flaky test

### PR DESCRIPTION
**Pull request by sarnex**
Closes: https://github.com/intel/llvm/issues/9008

### CODE REVIEW

File: sycl/test-e2e/Basic/memory-consumption.cpp [hunk 1]
```diff
@@ -7,6 +7,8 @@
 // extra memory allocation on device creation.
 // Issue #661: The test is failing sporadically on HIP AMD.
 // UNSUPPORTED: windows, opencl, hip_amd
+// Issue #9008: The test is failing sporadically on all hardware
+// UNSUPPORTED: *
```
Comment (bader) [inline sycl/test-e2e/Basic/memory-consumption.cpp:11]:
```suggestion
// REQUIRES: TEMPORARY_DISABLED
```

Comment (bso-intel) [reply sycl/test-e2e/Basic/memory-consumption.cpp:11]: I actually wanted to close the issue yesterday.
Did you find this test is still failing yesterday or today?

Comment (bader) [reply sycl/test-e2e/Basic/memory-consumption.cpp:11]: Today - https://github.com/intel/llvm/actions/runs/4689271741/jobs/8311985618

**Comment by sarnex** [comment]
@bader applied your feedback, we can prolly merge without ci (maybe?)

**Review by bso-intel — APPROVED** [review]
Thanks for the failure links.

I think this test is inherently flaky.
Just comparing memory usage before and after the SYCL kernel is not a good method to check the memory leak.
We don't know whether there is other thread / task is running in addition to the kernel.
So, this test is not very useful.

**Comment by sarnex** [comment]
@bso-intel thanks, if i made another PR after this to just delete the test would you support it?

**Comment by bso-intel** [comment]
> @bso-intel thanks, if i made another PR after this to just delete the test would you support it?

Yes, I would like to remove this test rather than disable it.
Unless we have a way to control OS from scheduling any other parallel task/thread running on the entire system, this test may pass today, it will come back and fail some day.

**Comment by sarnex** [comment]
ok ill do that next

File: sycl/test-e2e/Basic/memory-consumption.cpp [hunk 2]
```diff
@@ -7,6 +7,8 @@
 // extra memory allocation on device creation.
 // Issue #661: The test is failing sporadically on HIP AMD.
```
Comment (bso-intel) [inline sycl/test-e2e/Basic/memory-consumption.cpp:8]: It seems this test has been already identified "flaky" although it was on HIP AMD platform before.
Anyway, measuring memory consumption at two different times cannot check memory leak reliably.
Let's remove this test from our E2E test suite.

Comment (sarnex) [reply sycl/test-e2e/Basic/memory-consumption.cpp]: done in latest commit

**Review by bso-intel — APPROVED** [review]

**Comment by sarnex** [comment]
@intel/llvm-gatekeepers Hey all can we merge? The gen12 failure is not related and this fixes a CI problem hit in every PR and every postcommit run. Thx.

**Pull request merged**